maintenance/wikia/HttpsMigration/migrateCheckUser.php

```diff
@@ -0,0 +1,176 @@
+			false, false,	'd' );
+	}
+
+	public function execute() {
+		$saveChanges = $this->hasOption( 'saveChanges' );
+
+		$srcDB = wfGetDB( DB_SLAVE, [], 'wikiaglobal' );
+		$dstSlave = wfGetDB( DB_SLAVE, [], 'wikia' );
+		$dstMaster = wfGetDB( DB_MASTER, [], 'wikia' );
+		// this script would generate a ton of db logs, disable them
+		$zeroSampler =  new BernoulliTrial( 0.0 );
+		$srcDB->setSampler( $zeroSampler );
+		$dstSlave->setSampler( $zeroSampler );
+		$dstMaster->setSampler( $zeroSampler );
+
+		$logLastId = 0;
+		$logLastTimestamp = 0;
+		$logDuplicates = 0;
+		$logMigrated = 0;
+		$changesLastId = 0;
+		$changesLastTimestamp = 0;
+		$changesDuplicates = 0;
+		$changesMigrated = 0;
+
+		while(true) {
+			$res = $srcDB->select(
+				self::CU_LOG_TABLE,
+				'*',
+				"cul_id > $logLastId",
+				__METHOD__,
+				[ 'ORDER BY' => 'cul_id ASC', 'LIMIT' => self::BATCH_SIZE ] );
+			$count = $res->numRows();
+			WikiaLogger::instance()->info( 'Processing next batch',
+				[ 'rows' => $count , 'lastTimestamp' => $logLastTimestamp ]
+			);
+			if ($count <= 0) {
+				break;
+			}
+			foreach ( $res as $row ) {
+				$logLastId = $row->cul_id;
+				$logLastTimestamp = $row->cul_timestamp;
+				$duplicateRes = $dstSlave->select(
+					self::CU_LOG_TABLE,
+					'cul_id',
+					[
+						'cul_timestamp' => $row->cul_timestamp,
+						'cul_user' => $row->cul_user,
+						'cul_type' => $row->cul_type
+					],
+					__METHOD__
+				);
+				if ( $duplicateRes->numRows() >= 1 ) {
+					$logDuplicates += 1;
+				} else {
+					$logMigrated += 1;
+					if ( $saveChanges ) {
+						$dstMaster->insert(
+							self::CU_LOG_TABLE,
+							[
+								'cul_timestamp' => $row->cul_timestamp,
+								'cul_user' => $row->cul_user,
+								'cul_user_text' => $row->cul_user_text,
+								'cul_reason' => $row->cul_reason,
+								'cul_type' => $row->cul_type,
+								'cul_target_id' => $row->cul_target_id,
+								'cul_target_text' => $row->cul_target_text,
+								'cul_target_hex' => $row->cul_target_hex,
+								'cul_range_start' => $row->cul_range_start,
+								'cul_range_end' => $row->cul_range_end,
+							],
+							__METHOD__
+						);
+					}
+				}
+			}
+		}
+
+		while(true) {
+			$res = $srcDB->select(
+				self::CU_CHANGES_TABLE,
+				'*',
+				"cuc_id > $changesLastId",
+				__METHOD__,
+				[ 'ORDER BY' => 'cuc_id ASC', 'LIMIT' => self::BATCH_SIZE ] );
+			$count = $res->numRows();
+			WikiaLogger::instance()->info( 'Processing next batch',
+				[ 'rows' => $count , 'lastTimestamp' => $changesLastTimestamp ]
+			);
+			if ($count <= 0) {
+				break;
+			}
+			foreach ( $res as $row ) {
+				$changesLastId = $row->cuc_id;
+				$changesLastTimestamp = $row->cuc_timestamp;
+				$duplicateRes = $dstSlave->select(
+					self::CU_CHANGES_TABLE,
+					'cuc_id',
+					[
+						'cuc_timestamp' => $row->cuc_timestamp,
+						'cuc_user' => $row->cuc_user,
+						'cuc_type' => $row->cuc_type
+					],
```

Comment:
Two issues here:
* I don't think `cuc_timestamp, cuc_user, cuc_type` is unique in cu_changes
* also there may be a performance issue, there is no index combining all of those 3 columns.